tests/integration/lib/TokenAuthTest.php

```
@@ -37,114 +37,64 @@ abstract class TokenAuthTest extends BaseTest
-     *                          artifact that describes the HTTP request to be
-     *                          made, and, if $tokenType is 'valid_token', the
-     *                          output artifact that describes how to validate
-     *                          the HTTP response body.
-     * @param string $fileName the name of the JSON test input artifact file
-     *                         that describes the HTTP request to be made, and,
-     *                         if $tokenType is 'valid_token', the output
-     *                         artifact that describes how to validate the HTTP
-     *                         response body.
-     * @param string|null $testKey if the test artifact files contain multiple
-     *                             keys, a single key must be provided to
-     *                             indicate which test to run, and this key
-     *                             must be present in both the input and output
-     *                             test artifacts.
+     *                          (@see provideTokenAuthTestData()).
+     * @param array input @see BaseTest::requestAndValidateJson(). Must have
+     *                    additional required keys 'endpoint_type', whose value
+     *                    is either 'controller' or 'rest', and
+     *                    'authentication_type', whose value is either
+     *                    'token_optional' or 'token_required'.
+     * @param array $output @see BaseTest::requestAndValidateJson(). If
+     *                      $tokenType is not 'valid_token', this will be
+     *                      ignored and instead the relevant
+     *                      authentication/authorization error testing will be
+     *                      performed.
      * @return mixed the decoded JSON response body of the request.
      * @throws Exception if the role is unrecognized or there is an error
```

This line won't have an affect on the running of tests, but you'll want to either change this to `\Exception` or `use Exception;` at the top of the file ( Unless you have a new class `IntegrationTests\Exception` ).

---

Just something else I noticed, you may want to note that the function can throw an exception if an invalid value for `endpoint_type`, `$tokenType`, or `authentication_type` is provided.

---

Done.